modules/ding_campaign/ding_campaign.module

```diff
@@ -190,7 +190,7 @@ function ding_campaign_get_campaigns($context, $limit, $offset) {
         AND n.status = 1
         AND r.cid = d.cid
         AND r.cid = n.nid
-      ORDER BY d.weight DESC', array(':nid' => (int) $nid))
+      ORDER BY d.weight ASC', array(':nid' => (int) $nid))
```

Comment:
https://inlead.atlassian.net/browse/EASYOPAC-1101

Reply:
This seems legit, however I find it weird that this is not fixed in upstream.
Wonder if this is even used anymore?

Reply:
Find out if this module is used anymore after switching to **ding_campaign_plus**.
I think this module might be deactivated on our clients.

Reply:
The **ding_campaign** itself is enabled. There is this **ding_campaign_migrate** module to migrate campaigns from **ding_campaign** to **ding_campaign_plus**. It's also the one to disable and uninstall the **ding_campaign** module during the migrations process.
So I assume that either the migration did not run, or **ding_campaign** was enabled later again.
Both of **ding_campaign** and **ding_campaign_plus** are enabled now, checked on hedbib.

Reply:
```
[bbsbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[borbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Enabled        7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[bronbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[faxbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[gronbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[guldbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[hedbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Enabled        7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[herbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[holbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 Ding!                                Ding Campaign Plus (ding_campaign_plus)                                        Module  Enabled
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[jambib@prod7 public_html]$  drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[lembib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[midbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[naesbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[skanbib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Enabled        7.x-5.0.1

[oybib@prod7 public_html]$ drush pm-list | grep campaign
 Ding!                                Ding campaign (ding_campaign)                                                  Module  Not installed  7.x-2.5
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                                  Module  Not installed  7.x-5.0.1
```

Reply:
I guess its wrongly enabled on `borbib` and `hedbib`.

Reply:
Why `ding_campaign_plus` is not listed in your output?

Reply:
Other modules was left out for the sake of space. It would look like this:
```
[bronbib@prod7 public_html]$ drush pm-list|grep campaign
 Ding!                                Ding campaign (ding_campaign)                                         Module  Not installed  7.x-2.5
 Ding!                                Ding Campaign Plus (ding_campaign_plus)                               Module  Enabled
 Ding!                                Ding Campaign Plus Auto-generate (ding_campaign_plus_auto)            Module  Enabled
 Ding!                                Ding Campaign Plus Basic (ding_campaign_plus_basic)                   Module  Enabled
 Ding!                                Ding Campaign Plus Facets (ding_campaign_plus_facet)                  Module  Enabled
 Ding!                                Ding Campaign Plus Object (ding_campaign_plus_object)                 Module  Enabled
 Ding!                                Ding Campaign Plus Search (ding_campaign_plus_search)                 Module  Enabled
 easyOPAC                             Ding campaign migrate (ding_campaign_migrate)                         Module  Enabled        7.x-5.0.1
 easyOPAC                             Ding campaign mobile (ding_campaign_mobile)                           Module  Enabled        7.x-5.0.0
 easyOPAC                             Ding Campaign Plus Extra (ding_campaign_plus_extra)                   Module  Enabled        7.x-5.0.2
```

Reply:
Okay, so I assume that this changeset in `ding_campaign` is not needed too.

Reply:
If it's an improvement we can merge it, as the module might come back to live at a stager state. However, I find it weird that it didn't work as expected before tho'.

Reply:
Yes it looks like bug fix though. Considering that this module is currently not used it should not spoil anything.

Reply:
Agree.